hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/MergeIntoHoodieTableCommand.scala

```diff
@@ -432,6 +432,26 @@ case class MergeIntoHoodieTableCommand(mergeInto: MergeIntoTable) extends Hoodie
       false
     }
 
+    // validate that we can support partial updates
+    if (writePartialUpdates) {
+      if (hoodieCatalogTable.tableConfig.getBootstrapBasePath.isPresent) {
+        throw new UnsupportedOperationException("Partial updates are not supported for bootstrap tables.")
+      }
+
+      if (!hoodieCatalogTable.tableConfig.populateMetaFields()) {
+        throw new UnsupportedOperationException("Partial updates are not supported for virtual key tables.")
+      }
+
+      if (HoodieAvroUtils.containsUnsupportedTypesForFileGroupReader(fullSchema)) {
+        throw new UnsupportedOperationException("Partial updates are not supported for tables with enum columns")
+      }
+
+      if (parameters.getOrElse(DataSourceReadOptions.SCHEMA_EVOLUTION_ENABLED.key(),
```

Comment:
Could `SparkConfigUtils#containsConfigProperty` be used?

Reply:
No, we need to check the value and that only has a method for string configs.